src/docs/torque.md

```diff
@@ -348,6 +349,81 @@ Call(f, Undefined);
 return fastArray; // Type error: fastArray is invalid here.
 ```
 
+#### Enums
+
+Enumerations provide a means to define a set of constants and group them under a name similar to
+the enum classes in C++. A declaration is introduced by the `enum` keyword and adheres to the following
+syntactical structure:
+
+<pre><code class="language-grammar">EnumDeclaration :
+  <b>extern</b> <b>enum</b> IdentifierName ExtendsDeclaration<sub>opt</sub> ConstexprDeclaration<sub>opt</sub> <b>{</b> IdentifierName<sub>list+</sub> (<b>, ...</b>)<sub>opt</sub> <b>}</b>
+</code></pre>
+
+A basic example looks like this:
+
+```torque
+extern enum LanguageMode extends Smi {
+  kStrict,
+  kSloppy
+}
+```
+
+This declaration defines a new type `LanguageMode`, where the `extends` clause specifies the underlying
+type, that is the runtime type used to represent a value of the enum. In this example, this is `TNode<Smi>`,
+since this is what the type `Smi` `generates`. A `constexpr LanguageMode` converts to `LanguageMode`
+in the generated CSA files since no `constexpr` clause is specified on the enum to replace the default name.
+If the `extends` clause is omitted, Torque will generate only the `constexpr` version of the type. The `extern` keyword tells Torque that there is a C++ definition of this enum. Currently, only `extern` enums are supported.
+
+Torque generates a distinct type and constant for each of the enum's entries. Those are defined
+inside a namespace that matches the enum's name. Necessary specializations of `FromConstexpr<>` are
+generated to convert from the entry's `constexpr` types to the enum type. The value generated for an entry in the C++ files is `<enum-constexpr>::<entry-name>` where `<enum-constexpr>` is the `constexpr` name generated for the enum. In the above example, those are `LanguageMode::kStrict` and `LanguageMode::kSloppy`.
+
+To summarize, the above example is desugared into something that looks very similar to this:
```

Comment:
I'd consider leaving out the desugared code example. What about just mentioned where the output for enums is written to in gen/torque-generated.

Reply:
There is no distinct file where stuff is written for enums. They appear all over the place where they are used, mostly.